include/accessor.h

```diff
@@ -407,6 +407,24 @@ class accessor<DataT, Dims, Mode, target::host_task> : public detail::accessor_b
 
 	template <access_mode M = Mode>
 	inline std::enable_if_t<detail::access::mode_traits::is_producer(M), DataT&> operator[](const id<Dims>& index) const {
+		
+#if CELERITY_ACCESSOR_BOUNDARY_CHECK
+		if(m_oob_indices != nullptr) {
+			const id<Dims> all_true = detail::id_cast<Dims>(id<3>(true, true, true));
+			const bool is_within_bounds_lo = (index >= m_accessed_virtual_subrange.offset) == all_true;
+			const bool is_within_bounds_hi = (index < (m_accessed_virtual_subrange.offset + m_accessed_virtual_subrange.range)) == all_true;
+
+			if((!is_within_bounds_lo || !is_within_bounds_hi)) {
+				for(int d = 0; d < Dims; ++d) {
+					std::lock_guard<std::mutex>guard (m_oob_mutex);
```

Comment:
```suggestion
					std::lock_guard<std::mutex> guard (m_oob_mutex);
```
Should've been caught by CI, unfortunately the check was broken (fixed in #215).